### PR TITLE
Add description for the update_alg parameter of decision trees

### PR DESCRIPTION
**Pull request by atremblay**

### CODE REVIEW

File: container_files/public_html/doc/builtin/procedures/Classifier.md [hunk 1]
```diff
@@ -98,6 +98,30 @@ when a generic classification step is required.
 
 ![](%%jmlclassifier decision_tree)
 
+#### Parameter update_alg
+The parameter update_alg can take three different values: normal, gentle and prob.
```
Comment (guyd) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: `update_alg`

File: container_files/public_html/doc/builtin/procedures/Classifier.md [hunk 2]
```diff
@@ -98,6 +98,30 @@ when a generic classification step is required.
...
+Here is how they work. We'll work with a leaf node that contains 8 positives
```
Comment (guyd) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: positives -> positive
negatives -> negative

File: container_files/public_html/doc/builtin/procedures/Classifier.md [hunk 3]
```diff
@@ -98,6 +98,30 @@ when a generic classification step is required.
...
+and 2 negatives labels.
+
+##### prob
+It is the proportion of positive class, so simply #pos/(#pos + #neg).
```
Comment (guyd) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: class -> classes

File: container_files/public_html/doc/builtin/procedures/Classifier.md [hunk 4]
```diff
@@ -98,6 +98,30 @@ when a generic classification step is required.
...
+0.8 - 0.2 = 0.6 and 1 - 0.6 = 0.4
+
+Those scores are fed to a function, f, of the exponential family, and output
+f(0.6) - f(0.4)
+
+The output of f is unbounded and will return a score between -infinity and +infinity.
+
+##### gentle
+Also uses the margin, but with a different function, g, bounded between -1 and 1.
+In an ensemble it is recommended to use this value.
```
Comment (guyd) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: In an ensemble?

File: container_files/public_html/doc/builtin/procedures/Classifier.md (hunk 3 above)
Comment (mailletf) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: consider using latex formatting for formulas. you can look at the tf/idf function for an example of how to do it

File: container_files/public_html/doc/builtin/procedures/Classifier.md (hunk 4 above)
Comment (mailletf) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: to clarify the term ensemble, maybe add something like: In an ensemble, for example a bagged decision tree, ...

Comment (guyd) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: and link to wikipedia

**Comment by guyd** [comment]
+1

File: container_files/public_html/doc/builtin/procedures/Classifier.md [hunk 5]
```diff
@@ -98,6 +98,34 @@ when a generic classification step is required.
 
 ![](%%jmlclassifier decision_tree)
 
+#### Parameter `update_alg`
+The parameter update_alg can take three different values: normal, gentle and prob.
+'normal' and 'gentle' are described in more details in this 
+[paper](http://projecteuclid.org/download/pdf_1/euclid.aos/1016218223)
```
Comment (mailletf) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: my only minor comment here is I'd vote adopting a pattern when we cite papers of having the full reference (like Author bla, bla, year, conference) plus a link if we have it. The ref could be in another section right before the glz, like "See also" with the ref or something. but in any case, more than just the link

File: container_files/public_html/doc/builtin/procedures/Classifier.md [hunk 6]
```diff
@@ -98,6 +98,38 @@ when a generic classification step is required.
 
 ![](%%jmlclassifier decision_tree)
 
+#### Parameter `update_alg`
+The parameter update_alg can take three different values: normal, gentle and prob.
+'normal' and 'gentle'.
```
Comment (mailletf) [inline container_files/public_html/doc/builtin/procedures/Classifier.md]: typo?

**Comment by nicolaskruchten** [comment]
+1, this is a big improvement :) we can always fine-tune more later

**Comment by mailletf** [comment]
+1

**Pull request merged**